src/util/useComputedFromDependencies.coffee

```diff
@@ -0,0 +1,30 @@
+import {useRef} from 'react'
+import {isFunction, get} from './helpers'
+import usePrevious from './usePrevious'
+import useMemoized from './useMemoized'
+
+useComputedFromDependencies = ({
+  compute
+  dependencies
+  additionalResolvedDependencies = []
+  props
+}) ->
+  if dependencies?
+    if isFunction dependencies
+      prevProps = usePrevious props
+      computedValueRef = useRef()
+      changed = not prevProps? or dependencies prevProps, props
```

Comment:
Would this consider a case where `prevProps` and `props` are both `null` a change?

Reply:
Yes. But when we're dealing with actual component `props` (which we are in this case), they're guaranteed to be non-null on every render (will always be a props object). So we're only dealing with the initial render where `prevProps` isn't yet initialized